tests/nested/manual/fakeinstaller-real/task.yaml

```diff
@@ -0,0 +1,243 @@
+  snap pack --filename=pc.snap pc-gadget/ 
+
+  # get an updated kernel
+  snap download --basename=pc-kernel --channel="$version/edge" pc-kernel
+  uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR"
+  mv "${NESTED_ASSETS_DIR}"/pc-kernel_*.snap pc-kernel.snap
+
+  # prepare a classic seed
+  # TODO:
+  # - create pc-classic custom gadget
+  # - repacked snapd snap
+  # (should be as simple as adding "--snap=./local-gadget.snap ...")
+  gendeveloper1 sign-model < "$TESTSLIB"/assertions/developer1-22-classic-dangerous.json > my.model
+  snap prepare-image --classic \
+      --channel=edge \
+      --snap ./pc-kernel.snap \
+      --snap ./pc.snap \
+      my.model \
+      ./classic-seed
+  # make the seed label more predictable for fake-installer auto-mode
+  LABEL=classic
+  mv ./classic-seed/system-seed/systems/* ./classic-seed/system-seed/systems/"$LABEL"
+  cp -a ./classic-seed/system-seed/ /var/lib/snapd/seed
+
+  # we don't need the fakestore anymore
+  teardown_fake_store "$STORE_DIR"
+
+
+  # build the fake-installer snap
+  snap install snapcraft --candidate --classic
+  snap install lxd --candidate
+  snap set lxd waitready.timeout=240
+  lxd waitready
+  lxd init --auto
+  echo "Setting up proxy for lxc"
+  if [ -n "${http_proxy:-}" ]; then
+    lxd.lxc config set core.proxy_http "$http_proxy"
+  fi
+  if [ -n "${https_proxy:-}" ]; then
+    lxd.lxc config set core.proxy_https "$http_proxy"
+  fi
+  (cd "$TESTSLIB"/fakeinstaller && snapcraft)
+  FAKEINSTALLER_SNAP="$(find "$TESTSLIB"/fakeinstaller/ -maxdepth 1 -name '*.snap')"
+  echo "found $FAKEINSTALLER_SNAP"
+
+  # create new disk for the installer to work on and attach to VM
+  truncate --size=4G fake-disk.img
+  # TODO: XXX: fakeinstaller should work on a empty disk too, it errors
+  # right now with "device with name "/dev/vdc" is not a physical disk
+  echo "label: gpt" | sfdisk "fake-disk.img"
+
+  NESTED_PARAM_EXTRA="-drive file=$(pwd)/fake-disk.img,if=virtio,snapshot=off"
+  export NESTED_PARAM_EXTRA
+
+  # create a VM and mount a cloud image
+  tests.nested build-image classic
+
+  # TODO: nested classic images do not support secure boot today so
+  #       this will not work to test the secure boot installer. So for
+  #       now the workaround is to boot classic to create user/ssh
+  #       keys, shutdown down, convert disk from qcow2->raw and rename
+  #       from classic->core and use nested_start_core_vm (like below)
+  #
+  # start it so that cloud-init creates ssh keys and user
+  nested_start_classic_vm
+
+  # make sure classic image is bootable with snakeoil keys
+  # TODO: move to nested_create_classic_image
+  # XXX: use assets from gadget instead?
+  for s in BOOT/BOOTX64.EFI ubuntu/shimx64.efi; do
+      remote.exec "sudo cp -a /boot/efi/EFI/$s /tmp"
+      remote.exec "sudo chmod 755 /tmp/$(basename $s)"
+      remote.pull /tmp/"$(basename $s)" .
+      nested_secboot_sign_file "$(basename $s)" "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
```

Comment:
you can use:
tests.nested secboot-sign file "$(basename $s)" "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"